models/Schematic_DataModel.php

```diff
@@ -40,14 +40,21 @@ protected function defineAttributes()
      * Populate data model from yaml.
      *
      * @param string $yaml
+     * @param string $overrideYaml
      *
      * @return Schematic_DataModel
      */
-    public static function fromYaml($yaml)
+    public static function fromYaml($yaml, $overrideYaml)
     {
         $data = Yaml::parse($yaml);
+        $overrideData = Yaml::parse($overrideYaml);
+        if ($overrideData != null) {
+            $mergedData = array_replace_recursive($data, $overrideData);
```

Comment:
With Cmap, If each array has an element with the same string key value, the latter will overwrite the former (different from array_merge_recursive).